tests/variable/variable.cpp

```diff
@@ -285,5 +349,67 @@ TEST(Variable, modelInvalidUnitsName) {
     EXPECT_EQ("invalid name", u->getName());
 }
 
+TEST(Variable, modelTwoNamedInitialisedVariables) {
+    const std::string in = "valid_name";
```

Comment:
If you are to keep track of the name of the component, then you probably ought to do the same for the name of the variables, as well as for their value.

Note: as a result, something similar ought to be done for the other tests.